mat64/hogsvd_example_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/gonum/matrix/mat64"
+)
+
+func ExampleHOGSVD() {
+	// Perform an HOGSVD factorization on food production/consumption data for the
+	// three years 1990, 2000 and 2014.
+	//
+	// See Ponnapalli et al. doi:10.1371/journal.pone.0028072 and
+	// Alter at al. doi:10.1073/pnas.0530258100 for more details.
+	var gsvd mat64.HOGSVD
```

Comment:
Yeah, I'd really rather not. `hog` is not a good abbreviation (**h**igher **o**rder **g**eneralised) and `hogsvd` is too long - but it is a generalised singular value decomposition (all lower case). The two values are in clearly different scopes and if we consider other cases like `r` for any particular concrete `io.Reader` - not stored in an `io.Reader` interface value - we're quite happy with that.